osu.Game.Tests/Visual/Menus/TestAudioDucking.cs

```diff
@@ -0,0 +1,141 @@
+// Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the MIT Licence.
+// See the LICENCE file in the repository root for full licence text.
+
+using System;
+using NUnit.Framework;
+using osu.Game.Audio.Effects;
+using osu.Game.Overlays;
+
+namespace osu.Game.Tests.Visual.Menus
+{
+    public partial class TestSceneAudioDucking : OsuGameTestScene
+    {
+        [Test]
+        public void TestMomentaryDuck()
+        {
+            AddStep("duck momentarily", () => Game.MusicController.DuckMomentarily(1000));
+        }
+
+        [Test]
+        public void TestMultipleDucks()
+        {
+            IDisposable duckOp1 = null!;
+            IDisposable duckOp2 = null!;
+
+            double normalVolume = 1;
+
+            AddStep("get initial volume", () =>
+            {
+                normalVolume = Game.Audio.Tracks.AggregateVolume.Value;
+            });
+
+            AddStep("duck one", () =>
+            {
+                duckOp1 = Game.MusicController.Duck(new DuckParameters
+                {
+                    DuckVolumeTo = 0.5f,
+                });
+            });
+
+            AddUntilStep("wait for duck to complete", () => Game.Audio.Tracks.AggregateVolume.Value, () => Is.EqualTo(normalVolume * 0.5f).Within(float.Epsilon));
+
+            AddStep("duck two", () =>
+            {
+                duckOp2 = Game.MusicController.Duck(new DuckParameters
+                {
+                    DuckVolumeTo = 0.2f,
+                });
+            });
+
+            AddUntilStep("wait for duck to complete", () => Game.Audio.Tracks.AggregateVolume.Value, () => Is.EqualTo(normalVolume * 0.2f).Within(float.Epsilon));
+
+            AddStep("restore two", () => duckOp2.Dispose());
+            AddUntilStep("wait for restore to complete", () => Game.Audio.Tracks.AggregateVolume.Value, () => Is.EqualTo(normalVolume * 0.5f).Within(float.Epsilon));
+
+            AddStep("restore one", () => duckOp1.Dispose());
+            AddUntilStep("wait for restore to complete", () => Game.Audio.Tracks.AggregateVolume.Value, () => Is.EqualTo(normalVolume).Within(float.Epsilon));
+        }
+
+        [Test]
+        public void TestMultipleDucksReverseOrder()
+        {
+            IDisposable duckOp1 = null!;
+            IDisposable duckOp2 = null!;
+
+            double normalVolume = 1;
+
+            AddStep("get initial volume", () =>
+            {
+                normalVolume = Game.Audio.Tracks.AggregateVolume.Value;
+            });
+
+            AddStep("duck one", () =>
+            {
+                duckOp1 = Game.MusicController.Duck(new DuckParameters
+                {
+                    DuckVolumeTo = 0.5f,
+                });
+            });
+
+            AddUntilStep("wait for duck to complete", () => Game.Audio.Tracks.AggregateVolume.Value, () => Is.EqualTo(normalVolume * 0.5f).Within(float.Epsilon));
```

Comment:
`float.Epsilon` is far too precise for this? `0.01` should be sufficient I would think.

Reply:
sure, although not sure this changes much?

Reply:
Just felt weird seeing `float.Epsilon` used in general, not a show-stopper or something.